kong/nginx_templates/nginx_kong.lua

```diff
@@ -0,0 +1,261 @@
+lua_max_running_timers 4096;
+lua_max_pending_timers 16384;
+lua_shared_dict kong                5m;
+lua_shared_dict kong_db_cache       ${{MEM_CACHE_SIZE}};
+lua_shared_dict kong_db_cache_miss 12m;
+lua_shared_dict kong_locks          8m;
+lua_shared_dict kong_process_events 5m;
+lua_shared_dict kong_cluster_events 5m;
+lua_shared_dict kong_healthchecks   5m;
+lua_shared_dict kong_rate_limiting_counters 12m;
+> if database == "cassandra" then
+lua_shared_dict kong_cassandra      5m;
+> end
+lua_socket_log_errors off;
+> if lua_ssl_trusted_certificate then
+lua_ssl_trusted_certificate '${{LUA_SSL_TRUSTED_CERTIFICATE}}';
+> end
+lua_ssl_verify_depth ${{LUA_SSL_VERIFY_DEPTH}};
+
+# injected nginx_http_* directives
+> for _, el in ipairs(nginx_http_directives)  do
+$(el.name) $(el.value);
+> end
+
+init_by_lua_block {
+    Kong = require 'kong'
+    Kong.init()
+}
+
+init_worker_by_lua_block {
+    Kong.init_worker()
+}
+
+
+> if #proxy_listeners > 0 then
+upstream kong_upstream {
+    server 0.0.0.1;
+    balancer_by_lua_block {
+        Kong.balancer()
+    }
+> if upstream_keepalive > 0 then
+    keepalive ${{UPSTREAM_KEEPALIVE}};
+> end
+}
+
+server {
+    server_name kong;
+> for i = 1, #proxy_listeners do
+    listen $(proxy_listeners[i].listener);
+> end
+    error_page 400 404 408 411 412 413 414 417 494 /kong_error_handler;
+    error_page 500 502 503 504 /kong_error_handler;
+
+    access_log ${{PROXY_ACCESS_LOG}};
+    error_log ${{PROXY_ERROR_LOG}} ${{LOG_LEVEL}};
+
+    client_body_buffer_size ${{CLIENT_BODY_BUFFER_SIZE}};
+
+> if proxy_ssl_enabled then
+    ssl_certificate ${{SSL_CERT}};
+    ssl_certificate_key ${{SSL_CERT_KEY}};
+    ssl_protocols TLSv1.1 TLSv1.2 TLSv1.3;
+    ssl_certificate_by_lua_block {
+        Kong.ssl_certificate()
+    }
+
+    ssl_session_cache shared:SSL:10m;
+    ssl_session_timeout 10m;
+    ssl_prefer_server_ciphers on;
+    ssl_ciphers ${{SSL_CIPHERS}};
+> end
+
+> if client_ssl then
+    proxy_ssl_certificate ${{CLIENT_SSL_CERT}};
+    proxy_ssl_certificate_key ${{CLIENT_SSL_CERT_KEY}};
+> end
+
+    real_ip_header     ${{REAL_IP_HEADER}};
+    real_ip_recursive  ${{REAL_IP_RECURSIVE}};
+> for i = 1, #trusted_ips do
+    set_real_ip_from   $(trusted_ips[i]);
+> end
+
+    # injected nginx_proxy_* directives
+> for _, el in ipairs(nginx_proxy_directives) do
+    $(el.name) $(el.value);
+> end
+
+    location / {
+        default_type                     '';
+
+        set $ctx_ref                     '';
+        set $upstream_host               '';
+        set $upstream_upgrade            '';
+        set $upstream_connection         '';
+        set $upstream_scheme             '';
+        set $upstream_uri                '';
+        set $upstream_x_forwarded_for    '';
+        set $upstream_x_forwarded_proto  '';
+        set $upstream_x_forwarded_host   '';
+        set $upstream_x_forwarded_port   '';
+
+        rewrite_by_lua_block {
+            Kong.rewrite()
+
```

Comment:
Everything outside of this block is the standard template and doesn't need review.